teuthology/packaging.py

```diff
@@ -369,3 +381,231 @@ def get_package_version(remote, package):
         )
 
     return installed_ver
+
+
+def _get_config_value_for_remote(ctx, remote, config, key):
+    """
+    Look through config, and attempt to determine the "best" value to use
+    for a given key. For example, given:
+
+        config = {
+            'all':
+                {'branch': 'master'},
+            'branch': 'next'
+        }
+        _get_config_value_for_remote(ctx, remote, config, 'branch')
+
+    would return 'master'.
```

Comment:
I guess I don't understand the rule for "best" here.  I woudl have naively thought that 'branch' was better, because it was higher-level, and despite the name of the key being 'all', it's a subset of config.